### PR TITLE
[ISSUE #22] Improving codeCov for mqtt.common

### PR DESCRIPTION
**Pull request by YxAc**
codeCov for mqtt.common to improving #22

### CODE REVIEW

File: mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/NamespaceUtil.java [hunk 1]
```diff
@@ -24,45 +24,35 @@ public class NamespaceUtil {
...
     public static String encodeToNamespaceResource(String namespace, String resource) {
-        return resource != null && namespace != null ? StringUtils.join(new String[]{namespace, "%", resource}) : resource;
+        return resource != null && namespace != null ? StringUtils.join(namespace, NAMESPACE_SPLITER, resource) : resource;
     }
 
     public static String decodeOriginResource(String resource) {
-        if (resource != null && resource.contains("%")) {
-            int firstIndex = resource.indexOf("%");
+        if (resource != null && resource.contains(NAMESPACE_SPLITER)) {
+            int firstIndex = resource.indexOf(NAMESPACE_SPLITER);
```
Comment (YxAc) [inline mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/NamespaceUtil.java:33]: using defined constant

File: mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/NamespaceUtil.java [hunk 2]
```diff
@@ -24,45 +24,35 @@ public class NamespaceUtil {
...
     }
 
     public static String decodeStoreNamespaceIdFromTopic(String topic) {
-        if (topic != null && topic.contains("%")) {
-            String storeNamespaceId = topic.split("%")[0];
-            return storeNamespaceId;
-        } else {
-            return null;
-        }
+        return splitNamespaceStr(topic);
```
Comment (YxAc) [inline mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/NamespaceUtil.java:48]: encapsulate duplicate code block

File: mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/StatUtil.java [hunk 3]
```diff
@@ -227,40 +227,7 @@ public static void addInvoke(String key, int num, long rt, boolean success) {
...
-        }
-
-        invoke.sumRt.addAndGet(rt);
-        if (invoke.maxRt.get() < rt) {
-            invoke.maxRt.set(rt);
-        }
-        if (invoke.minRt.get() > rt) {
-            invoke.minRt.set(rt);
-        }
+        addInvoke(key, 1, rt, success);
```
Comment (YxAc) [inline mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/StatUtil.java:230]: reusing 'addInvoke' method

File: mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java [hunk 4]
```diff
@@ -31,7 +31,7 @@ public class TopicUtils {
      * @return
      */
     public static String normalizeTopic(String topic) {
-        if (topic == null) {
+        if (topic == null || topic.isEmpty()) {
```
Comment (YxAc) [inline mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java:34]: bug of 'empty str'

File: mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java [hunk 5]
```diff
@@ -93,10 +93,10 @@ public static boolean isP2pTopic(String topic) {
     }
 
     public static String getP2Peer(MqttTopic mqttTopic, String namespace) {
-        if (!isP2P(mqttTopic.getSecondTopic())) {
+        if (mqttTopic.getSecondTopic() == null || mqttTopic.getFirstTopic() == null) {
             return null;
         }
-        if (mqttTopic.getSecondTopic() == null || mqttTopic.getFirstTopic() == null) {
+        if (!isP2P(mqttTopic.getSecondTopic())) {
```
Comment (YxAc) [inline mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java:99]: reserve the order of 'null judgment' and 'p2p judgment'

**Comment by YxAc** [comment]
Wonder under what conditions the codecov-commenter will be triggered？ @tianliuliu , thanks :-)

**Comment by tianliuliu** [comment]
> Wonder under what conditions the codecov-commenter will be triggered？ @tianliuliu , thanks :-)

it can be triggered always, sometimes it dosen't work,  maybe the bug itself.

File: mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java (hunk 4 above)
Comment (tianliuliu) [reply mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java:34]: may be use StringUtils.isBlank(topic) will be better

Comment (YxAc) [reply mqtt-common/src/main/java/org/apache/rocketmq/mqtt/common/util/TopicUtils.java]: > may be use StringUtils.isBlank(topic) will be better

done

**Review by tianliuliu — APPROVED** [review]

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**